lib/pbench/server/unpack_tarballs.py

```diff
@@ -0,0 +1,135 @@
+import glob
+from logging import Logger
+import os
+from pathlib import Path
+import tempfile
+from typing import NamedTuple
+
+from pbench.server import PbenchServerConfig
+from pbench.server.filetree import FileTree
+from pbench.server.report import Report
+from pbench.server.utils import get_tarball_md5
+
+
+class Results(NamedTuple):
+    total: int
+    success: int
+
+
+class Unpack:
+    LINKSRC = "TO-UNPACK"
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        """Unpacks Tarball in the INCOMING Directory
+
+        Args:
+            config: PbenchServerConfig configuration object
+            logger: A Pbench python Logger
+        """
+        self.config = config
+        self.logger = logger
+
+    def unpack(self, lb_args: int, ub_args: int) -> Results:
```

Comment:
Name mismatch: `erver/bin/pbench-unpack-tarballs.py` calls `unpack_obj.unpack_tarball`, not `unpack`. And as I mentioned there, I think it should `unpack_tarballls` (or maybe `unpack_tar_balls`:  I think there is a precedent of saying `tar ball`, rather than `tarball` in much (if not all) of the code).

Reply:
By default, `ub_args` is passed as a `float("inf")` which is probably going to cause problems here - maybe?

Reply:
Passing a floating point infinity or any other floating point number to a function which expects and integer would cause problems...except that we're programming in Python which basically doesn't care about those sorts of details.  😋

But, from an orthodoxy point of view, perhaps we should consider reforming the type hint on the upper bound to indicate that it could be a `float` in addition to an `int`.